### DISCUSSION TITLE
GCP functions error : To use middleware you must provide a `hostname` and `port` to the Next.js Server

### DISCUSSION THREAD

**Discussion by janfabian** [vercel/next.js#37685] [Help]
Hi, I have my nextjs app deployed on GCP Cloud run (via Firebase), do you know what should I use for the properties `hostname` and `port` in order to be able to call middleware? 

My current nextjs custom server

```import { onRequest } from "firebase-functions/v2/https";
import next from "next";
import config from "./next.config.cjs";

const app = next({
  conf: { ...config, distDir: "nextjs/.next", hostname: ???, port: ???},
});
const handle = app.getRequestHandler();

export const server = onRequest(
  {
    region: "us-central1",
    memory: "1GiB",
  },
  (request, response) => {
    return app.prepare().then(() => handle(request, response));
  }
);
```

**Comment by icyJoseph** [comment]
Could you try with `hostname` set to `0.0.0.0` or `localhost` and port I guess, `process.env.PORT`? or maybe just 3000. https://nextjs.org/docs/advanced-features/custom-server
